torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -630,7 +630,7 @@ def cuda(
         return st_cuda
 
     def to(self, *args, **kwargs) -> ShardedTensor:
-        current_device = self._local_shards[0].tensor.device
+        current_device = self._local_shards[0].tensor.device if self._local_shards else torch.cuda.current_device()
```

Comment:
nit: looks like lint fails, I think it's because `torch.cuda.current_device` returns an int. We should wrap it with `torch.device(torch.cuda.current_device)` to fix the lint.

Also, the ShardedTensor might not always be a "cuda" device as always, so when local shards is empty, we shall check if `self.pg._get_backend_name() == "gloo"`, if so, we mark current_device as cpu, else we use `torch.cuda.current_device()`?